phpseclib/File/X509.php

```diff
@@ -4614,4 +4614,27 @@ function _extractBER($str)
         $temp = preg_match('#^[a-zA-Z\d/+]*={0,2}$#', $temp) ? base64_decode($temp) : false;
         return $temp != false ? $temp : $str;
     }
+
+    /**
+     * Returns the OID corresponding to a name
+     *
+     * What's returned in the associative array is either a name or an OID if no OID to name mapping is available.
+     * The problem with this is that what may be an unmapped OID in one version of phpseclib may not be unmapped
+     * in the next version, so apps that are looking at this OID may not be able to work from version to version.
+     *
+     * This method will return the OID if a name is passed to it and if no mapping is avialable it'll assume that
+     * what's being passed to it already is an OID and return that instead. A few examples.
```

Comment:
> If you work on IDs, you don't have to translate to names. If you work on names, you don't handle IDs anyway, so returning an ID is pointless.

I don't agree. Presumably you should always want to work on names over IDs as they're more human readable. But sometime's a name just isn't going to be available and in that case you don't have a choice - you have to work with an ID. But, with the way phpseclib currently works, although `$cert['tbsCertificate']['extensions'][0]['extnId']` may be an ID in v1.0.0 it might not be an ID in v1.0.1 - it might be a name.

Per your proposal I'd rename `getOID`, as implemented in this PR, to `mapOID` and then implement `getOID` such that it returns null if the string that's passed to it doesn't exist as  a name?

Personally, I'm thinking `getOID` is a better function name for the function as implemented in this PR. `getOID` doesn't imply that a mapping exists whereas `mapOID` does...